src/x/os/limits_other.go

```diff
@@ -42,3 +45,9 @@ func CanGetProcessLimits() (bool, string) {
 func GetProcessLimits() (ProcessLimits, error) {
 	return ProcessLimits{}, errUnableToDetermineProcessLimits
 }
+
+// RaiseProcessNoFileToNROpen attempts to raise the process num files
+// open limit to the nr_open system value.
+func RaiseProcessNoFileToNROpen() (RaiseProcessNoFileToNROpenResult, error) {
```

Comment:
If we're gonna start doing this it might be nice to have a generic `AttemptToSetProcessLimits` which calls this function. That way we don't have to change `server.go` each time and we have an established pattern

Reply:
This is that kind of method however, it is meant to be abstract and returns enough information (without taking a logger itself).  That way callers can either log it as a warning or a hard error, depending on their situation.